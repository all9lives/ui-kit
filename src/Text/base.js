import { css } from 'styled-components'
import Unit from '../utils/Unit'
import Size from '../utils/Size'
import Color from '../utils/Color'

const base = css`
  font-size: ${props => Unit.convert(Size[props.size] || props.size)};
  color: ${props => (Color[props.color] || props.color)};
  margin: ${props => Unit.parse(props.margin, (margin) => margin * Size.spacing)};
  align-self: ${props => props.alignSelf};
`

export default base
