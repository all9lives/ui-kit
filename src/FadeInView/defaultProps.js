export default {
  delay: 1000,
  duration: 300
}
