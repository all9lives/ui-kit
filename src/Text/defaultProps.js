import Color from '../utils/Color'
import Unit from '../utils/Unit'
import Size from '../utils/Size'

export default {
  size: Unit.convert(Size.regular),
  color: Color.black
}
