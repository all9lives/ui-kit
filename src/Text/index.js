import React from 'react';
import { Text as PrimitiveText } from 'react-primitives';

const Text = ({ children }) => (
  <PrimitiveText>{children}</PrimitiveText>
);

export default Text;
