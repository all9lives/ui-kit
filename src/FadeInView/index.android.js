import React, { Component } from 'react'
import { Animated } from 'react-native'
import defaultProps from './defaultProps'

class FadeInView extends Component {
  static defaultProps = defaultProps
  animatedOpacity = new Animated.Value(0)
  componentDidMount () {
    const { delay, duration } = this.props
    this.timeout = setTimeout(() => {
      Animated.timing(this.animatedOpacity, {
        toValue: 1,
        duration
      }).start()
    }, delay)
  }
  componentWillUnmount () {
    clearTimeout(this.timeout)
  }
  render () {
    const { children } = this.props
    return (
      <Animated.View style={{ opacity: this.animatedOpacity }}>
        {children}
      </Animated.View>
    )
  }
}

export default FadeInView
