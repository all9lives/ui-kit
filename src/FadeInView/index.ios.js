import FadeInView from './index.android'

export default FadeInView
