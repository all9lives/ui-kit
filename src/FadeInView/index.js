import React, { Component } from 'react'
import defaultProps from './defaultProps'

class FadeInView extends Component {
  static defaultProps = defaultProps
  state = {
    animated: false
  }
  componentDidMount () {
    const { delay } = this.props
    this.timeout = setTimeout(() => {
      this.setState({ animated: true })
    }, delay)
  }
  componentWillUnmount () {
    clearTimeout(this.timeout)
  }
  render () {
    const { children, duration } = this.props
    return (
      <div style={{ opacity: this.state.animated ? 1 : 0, transition: `opacity ${duration}ms linear` }}>
        {children}
      </div>
    )
  }
}

export default FadeInView
