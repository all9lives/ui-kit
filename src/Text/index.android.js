import styled from 'styled-components/native'
import base from './base'
import defaultProps from './defaultProps'
import Size from '../utils/Size'

const getLineHeight = (props) => {
  if (props.content) {
    return Size.spacing * 2.5
  }
  if (props.size === 'title') {
    return Size.spacing * 3
  }
  return Size.spacing * 2
}

const Text = styled.Text`
  ${base}
  line-height: ${props => getLineHeight(props)};
`
Text.defaultProps = defaultProps

export default Text
