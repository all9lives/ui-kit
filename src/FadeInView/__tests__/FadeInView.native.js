/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import FadeInView from '../index.ios.js'
import Text from '../../Text'

jest.mock('Animated', () => {
  const ActualAnimated = require.requireActual('Animated')
  return {
    ...ActualAnimated,
    timing: (value, config) => {
      return {
        start: (callback) => {
          value.setValue(config.toValue)
          callback && callback()
        }
      }
    }
  }
})

describe('FadeInView.native', () => {
  it('renders a snapshot', () => {
    const wrapper = shallow(
      <FadeInView>
        <Text>Apple</Text>
      </FadeInView>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `delay:3000`', () => {
    const wrapper = shallow(
      <FadeInView delay={3000}>
        <Text>Apple</Text>
      </FadeInView>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `duration:1000`', () => {
    const wrapper = shallow(
      <FadeInView duration={1000}>
        <Text>Apple</Text>
      </FadeInView>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot after delay', () => {
    jest.useFakeTimers()
    const wrapper = shallow(
      <FadeInView>
        <Text>Apple</Text>
      </FadeInView>
    )
    jest.runAllTimers()
    wrapper.update()
    expect(wrapper).toMatchSnapshot()
  })
})
