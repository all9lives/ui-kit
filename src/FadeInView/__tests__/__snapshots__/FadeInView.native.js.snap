// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FadeInView.native renders a snapshot 1`] = `
<AnimatedComponent
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <Styled(Text)
    alignSelf="auto"
    color="#191919"
    margin={0}
    size={16}
  >
    Apple
  </Styled(Text)>
</AnimatedComponent>
`;

exports[`FadeInView.native renders a snapshot after delay 1`] = `
<AnimatedComponent
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Styled(Text)
    alignSelf="auto"
    color="#191919"
    margin={0}
    size={16}
  >
    Apple
  </Styled(Text)>
</AnimatedComponent>
`;

exports[`FadeInView.native renders a snapshot with \`delay:3000\` 1`] = `
<AnimatedComponent
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <Styled(Text)
    alignSelf="auto"
    color="#191919"
    margin={0}
    size={16}
  >
    Apple
  </Styled(Text)>
</AnimatedComponent>
`;

exports[`FadeInView.native renders a snapshot with \`duration:1000\` 1`] = `
<AnimatedComponent
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <Styled(Text)
    alignSelf="auto"
    color="#191919"
    margin={0}
    size={16}
  >
    Apple
  </Styled(Text)>
</AnimatedComponent>
`;
